src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncEHTests.cs

```diff
@@ -1229,6 +1229,75 @@ public static void Main()
             CompileAndVerify(source, expectedOutput: expected);
         }
 
+        [WorkItem(74, "https://github.com/dotnet/roslyn/issues/1334")]
+        [Fact]
+        public void AsyncInCatchRethrow01()
+        {
+            var source = @"
+using System;
+using System.Threading.Tasks;
+
+class Test
+{
+    static async Task<int> F()
+    {
+        await Task.Yield();
+        return 2;
+    }
+
+    static async Task<int> G()
+    {
+        int x = 0;
+
+        try
+        {
+            try
+            {
+                x = x / x;
+            }
+            catch (ArgumentNullException)
+            {
+                x = await F();
```

Comment:
Would it be interesting to test what re-throw would do after this statement? Test two or more catch blocs with await and re-throw?